github/release_notes/util.py

```diff
@@ -172,239 +176,187 @@ def release_note_blocks(self):
-    # included
-    # pylint: disable=undefined-variable
-    tags = _ \
-        .chain(repo.tags) \
-        .map(lambda tag: {"tag": tag.name, "commit": tag.commit.hexsha}) \
-        .filter(lambda item: _.find(release_tags, lambda el: el == item['tag'])) \
-        .filter(lambda item: is_valid_semver(item['tag'])) \
-        .key_by('commit') \
-        .map_values('tag') \
-        .value()
-    # pylint: enable=undefined-variable
-    return tags
-
-
-def reachable_release_tags_from_commit(
-    github_helper: GitHubRepositoryHelper,
-    repo: git.Repo,
-    commit: git.objects.Commit
-) -> [str]:
-    tags = release_tags(github_helper, repo)
-
-    visited = set()
-    queue = list()
-    queue.append(commit)
-    visited.add(commit.hexsha)
-
-    reachable_tags = list()
-
-    while queue:
-        commit = queue.pop(0)
-        if commit.hexsha in tags:
-            reachable_tags.append(tags[commit.hexsha])
-        not_visited_parents = _.filter(commit.parents,
-            lambda parent_commit: parent_commit.hexsha not in visited
+    def _rls_note_objs(
+        self,
+        repository_branch: str=None,
+        commit_range: str=None,
+    ) -> typing.List[ReleaseNote]:
+        if not commit_range:
+            commit_range = self.calculate_range(
+                repository_branch,
+            )
+        info(f'Fetching release notes from revision range: {commit_range}')
+        commits = self.commits_in_range(
+            commit_range=commit_range,
+            repository_branch=repository_branch,
         )
-        if not_visited_parents:
-            queue.extend(not_visited_parents)
-            visited |= set(_.map(not_visited_parents, lambda commit: commit.hexsha))
-
-    reachable_tags.sort(key=lambda t: version.parse_to_semver(t), reverse=True)
-
-    if not reachable_tags:
-        warning('no release tag found, falling back to root commit')
-        root_commits = repo.iter_commits(rev=commit, max_parents=0)
-        root_commit = next(root_commits, None)
-        if not root_commit:
-            fail('could not determine root commit from rev {rev}'.format(rev=commit.hexsha))
-        if next(root_commits, None):
-            fail(
-                'cannot determine range for release notes. Repository has multiple root commits. '
-                'Specify range via commit_range parameter.'
+        pr_numbers = fetch_pr_numbers_from_commits(commits=commits)
+        verbose(f'Merged pull request numbers in range {commit_range}: {pr_numbers}')
+        release_note_objs = self.fetch_release_notes_from_prs(
+            pr_numbers_in_range=pr_numbers,
+        )
+        release_note_objs.extend(
+            fetch_release_notes_from_commits(
+                commits=commits,
+                current_component=self.component,
             )
-        reachable_tags.append(root_commit.hexsha)
-
-    return reachable_tags
-
-
-def commits_in_range(
```

Comment:
why is there this huge diff? judging from the signature, I would have thought that this function does not need to be touched at all?